java/src/main/java/com/powsybl/dataframe/network/AbstractNetworkDataframeMapper.java

```diff
@@ -38,6 +43,35 @@ public void createDataframe(Network network, DataframeHandler dataframeHandler,
         mappers.stream().forEach(mapper -> mapper.createSeries(items, dataframeHandler));
     }
 
+    protected List<T> getFilteredItems(Network network, DataframeFilter dataframeFilter) {
+        if (dataframeFilter.getRowsIds().size() == 0) {
+            return getItems(network);
+        }
+        return getItems(network).stream().filter(item -> filterItem(item, dataframeFilter)).collect(Collectors.toList());
```

Comment:
I wonder what would be the perf impact on doing a search by IDs instead of a filter, for small selections:
do you think you could make a small test for this?

Like trying to get 1 switch by its ID with this approach on one side, and with a "network.getSwitch(id)" approach on the other side, and time both approaches, to check if it can be worth it to do something "smart" about it in the future or not ? (not necessarily in that PR)

Reply:
searching by id (a single switch id) is roughly 30 times faster than applying a filter
(360 times faster than a pandas filter applied on the whole elements set, python-side). 
The latest commit contains the search-by-ids  implementation.